### DISCUSSION TITLE
How to handle Skip Action

### DISCUSSION THREAD

**Discussion by arishanapalli** [ephread/Instructions#291] [Q&A]
How to handle skip action.

If user clip skip, I want to show alert screen -> do you want to skip the Instructions screen?  Accept / No -> if click Accept  then skip the instructions.

**Comment by ephread** [comment]
Changing the behaviour of skip button is not possible. The next best options would be to not set `skipView` on `CoachMarksController` and instead attach a custom button as an ornament, by implementing `CoachMarksControllerDelegate.coachMarksController(_:configureOrnamentsOfOverlay:)`. For instance:

```swift
func coachMarksController(_ coachMarksController: CoachMarksController,
                          configureOrnamentsOfOverlay overlay: UIView) {
    let button = UIButton()
    button.setTitle("Skip", for: .normal)
    button.translatesAutoresizingMaskIntoConstraints = false
    button.addTarget(self, action: #selector(didTapSkipButton), for: .touchUpInside)

    overlay.addSubview(button)

    button.trailingAnchor.constraint(equalTo: overlay.trailingAnchor).isActive = true
    button.topAnchor.constraint(equalTo: overlay.topAnchor, constant: 100).isActive = true
}

@objc didTapSkipButton() {
    // Present Alert.
}
```
